apps/controllo-ore-x/src/app/_shared/components/sidenav/sidenav.component.ts

```diff
@@ -31,7 +50,9 @@ export class SidenavComponent implements OnInit, SubscriptionsLifecycle {
       this._sidenavService.visibiliyObservable.subscribe(
         (isOpen) => (this.isSidenavOpen = isOpen),
       ),
+      this._router.events.subscribe(() => {
+        this.selectedSection = this._router.url.split('/')[2];
```

Comment:
```suggestion
        this.selectedSection = this.getActiveSectionFromUrl(this._router_url)
```

Extracting this into its own function makes it possible to maintain only that specific function in the future without having to break this subscription. as of now the function will just be `return url.split(etc.)`, but in the future it could become more complex and having it in a special function also creates a good ground for that.